cmd/oras/root/attach.go

```diff
@@ -87,10 +87,22 @@ Example - Attach file to the manifest tagged 'v1' in an OCI image layout folder
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			opts.RawReference = args[0]
 			opts.FileRefs = args[1:]
-			if err := option.Parse(cmd, &opts); err != nil {
-				return err
+			err := option.Parse(cmd, &opts)
+			if err == nil {
+				if opts.Reference != "" {
+					return nil
+				}
+				// ensure reference is not empty
+				err = oerrors.NewErrEmptyTagOrDigest(opts.RawReference, cmd, true)
```

Comment:
nit: I just noticed this block of code is `EnsureReferenceNotEmpty` that you have removed. Consider
```suggestion
				err = opts.EnsureReferenceNotEmpty(cmd, true)
				if err == nil {
					return nil
				}
```
The method name is self-explainable.